src/app/app.component.ts

```diff
@@ -12,11 +13,14 @@ export class AppComponent {
 
     constructor(
         private _gateway: GatewayService,
-        private _postService: PostService
+        private _postService: PostService,
+        private _subredditService: SubredditService,
     ) {
-        this._gateway.fetchPosts()
-            .subscribe(response => {
-                this._postService.setPosts(response.posts);
-            })
+        this._gateway.fetchPosts().subscribe((response) => {
+            this._postService.setPosts(response);
+        });
+        this._gateway.fetchSubreddits().subscribe((response) => {
+            this._subredditService.setSubreddits(response);
+        });
```

Comment:
Is it neccessary to have this here? That mean that we are fetching posts and subreddits every time even if there are no components to show them. Like for landing page/registration form/ (future) login form.

Reply:
No, it was before authorizedComponent was created and I forgot to solve this problem. Now it's in the AuthorizedComponent